### PR TITLE
rxr: udp socket size max of INT_MAX/2

### PR DESCRIPTION
**Pull request by thattil**

### CODE REVIEW

File: slsReceiverSoftware/src/Implementation.cpp [hunk 1]
```diff
@@ -962,9 +962,14 @@ void Implementation::setUDPSocketBufferSize(const int64_t s) {
     for (unsigned int i = 0; i < listSize; ++i) {
```
Comment (erikfrojdh) [inline slsReceiverSoftware/src/Implementation.cpp:962]: This could be size_t

Comment (erikfrojdh) [reply slsReceiverSoftware/src/Implementation.cpp:962]: actually when I think about it a range based for loop is even better:
```cpp
for (auto& l : listener)
    l.createDummy...
```

File: slsReceiverSoftware/src/Implementation.cpp [hunk 2]
```diff
@@ -946,12 +946,12 @@ void Implementation::setUDPPortNumber2(const uint32_t i) {
...
 
-int64_t Implementation::getUDPSocketBufferSize() const {
+int Implementation::getUDPSocketBufferSize() const {
     return udpSocketBufferSize;
 }
 
-void Implementation::setUDPSocketBufferSize(const int64_t s) {
-    int64_t size = (s == 0) ? udpSocketBufferSize : s;
+void Implementation::setUDPSocketBufferSize(const int s) {
+    int size = (s == 0) ? udpSocketBufferSize : s;
```
Comment (erikfrojdh) [inline slsReceiverSoftware/src/Implementation.cpp:954]: This line might need a comment or rewrite?

**Review by erikfrojdh — CHANGES_REQUESTED** [review]
Minor stuff in the setting function

File: slsReceiverSoftware/src/ClientInterface.cpp [hunk 3]
```diff
@@ -1138,13 +1139,20 @@ int ClientInterface::get_additional_json_header(Interface &socket) {
 }
 
 int ClientInterface::set_udp_socket_buffer_size(Interface &socket) {
-    auto index = socket.Receive<int64_t>();
-    if (index >= 0) {
+    auto index = socket.Receive<int>();
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsReceiverSoftware/src/ClientInterface.cpp:1142]: done

File: slsReceiverSoftware/src/Implementation.cpp (hunk 1 above)
Comment (thattil) [reply slsReceiverSoftware/src/Implementation.cpp:962]: done

File: slsReceiverSoftware/src/Implementation.cpp (hunk 2 above)
Comment (thattil) [reply slsReceiverSoftware/src/Implementation.cpp:954]: done

**Pull request merged**